tests/test_particlesets.py

```diff
@@ -373,6 +385,73 @@ def AddLat(particle, fieldset, time):
     assert np.allclose([p.lat - n * 0.1 for p in pset], np.zeros(npart), rtol=1e-12)
 
 
+def setup_pset_execute(*, fieldset: FieldSet, outputdt: timedelta, execute_kwargs, particle_class=ScipyParticle):
+    # fieldset = create_fieldset_zeros_simple()
+    npart = 10
+
+    if fieldset is None:
+        fieldset = create_fieldset_zeros_simple()
+
+    pset = ParticleSet(
+        fieldset,
+        pclass=particle_class,
+        lon=np.full(npart, fieldset.U.lon.mean()),
+        lat=np.full(npart, fieldset.U.lat.mean()),
+    )
+
+    with tempfile.TemporaryDirectory() as dir:
+        name = f"{dir}/test.zarr"
+        output_file = pset.ParticleFile(name=name, outputdt=outputdt)
+
+        pset.execute(DoNothing, output_file=output_file, **execute_kwargs)
+        ds = xr.open_dataset(name, engine="zarr").load()
+
+    return ds
+
+
+def test_pset_execute_outputdt_forwards():
+    """Testing output data dt matches outputdt in forward time."""
+    outputdt = timedelta(hours=1)
+    runtime = timedelta(hours=5)
+    dt = timedelta(minutes=5)
+
+    ds = setup_pset_execute(
+        fieldset=create_fieldset_zeros_simple(), outputdt=outputdt, execute_kwargs=dict(runtime=runtime, dt=dt)
+    )
+
+    assert np.all(ds.isel(trajectory=0).time.diff(dim="obs").values == np.timedelta64(outputdt))
+
+
+def test_pset_execute_outputdt_backwards():
+    """Testing output data dt matches outputdt in backwards time."""
+    outputdt = timedelta(hours=1)
+    runtime = timedelta(days=2)
+    dt = -timedelta(minutes=5)
+
+    ds = setup_pset_execute(
+        fieldset=create_fieldset_zeros_simple(), outputdt=outputdt, execute_kwargs=dict(runtime=runtime, dt=dt)
+    )
+    file_outputdt = ds.isel(trajectory=0).time.diff(dim="obs").values
+    assert np.all(file_outputdt == np.timedelta64(-outputdt))
+
+
+def test_pset_execute_outputdt_backwards_fieldset_timevarying():
+    """test_pset_execute_outputdt_backwards() still passed despite #1722 as it doesn't account for time-varying fields,
+    which for some reason #1722
+    """
+    outputdt = timedelta(hours=1)
+    runtime = timedelta(days=2)
```

Comment:
I encountered a bit of difficulty writing these tests as even here changing `runtime` to `days=1` made the bug go away.

@willirath did you also encounter some difficulty writing your minimal example?

Reply:
> @willirath did you also encounter some difficulty writing your minimal example?

No, I basically posted exactly the code that let me discover the bug. So maybe I was just "lucky".